[{Ong Wen Pin}] iP

ongwenpin opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

# 

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ongwenpin committed:
Added Greet, Echo and Exit to Duke

ongwenpin committed:
Duke can now add things to list

ongwenpin committed:
Duke can now mark task as done and unmark it

ongwenpin committed:
Duke can track 3 types of tasks

ongwenpin committed:
Did Automated Text UI Testing

ongwenpin committed:
Duke can now handle errors

ongwenpin committed:
Duke can now delete task from list

ongwenpin committed:
Allow Duke to save task

ongwenpin committed:
Add date and time to deadlines for Duke

ongwenpin committed:
Merge branch 'branch-Level-7'

ongwenpin committed:
Merge branch-Level-8 to master

ongwenpin committed:
Refactor the code to make it more OOP

ongwenpin committed:
Organise all the classes into packages

ongwenpin committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

ongwenpin committed:
Add gradle to automate project build

ongwenpin committed:
Add JUnit tests to test behavior of the code

ongwenpin committed:
Add JavaDoc comments

ongwenpin committed:
Tweak the code to comply with a coding standard

ongwenpin committed:
Add a command to allow user to find tasks through keyword

ongwenpin committed:
Merge branch 'branch-A-JavaDoc'

ongwenpin committed:
Merge branch 'branch-A-CodingStandard'

ongwenpin committed:
Merge branch 'branch-Level-9'

ongwenpin committed:
Added a command to allow user to find task using keyword

ongwenpin committed:
Use checkstyle to fix some coding style violations

Donovan9617 commented on src/main/java/duke/parse/Parser.java:
```diff
@@ -0,0 +1,119 @@
+        Command commandToReturn = null;
+        String[] inputSpilt = input.split(" ", 2);
+        if (inputSpilt.length == 2) {
+            if (inputSpilt[0].equals(MarkCommand.COMMAND_WORD)) {
```
As inputSpilt[0] is used many times, perhaps it is good to give it a variable name for easier reference?

ongwenpin replied:
you're right, thank you for the suggestion

Donovan9617 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,43 @@
+        this.description = description;
+        this.isDone = false;
+    }
+
+    protected String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    protected String getSavedStatusIcon() {
+        return (isDone ? "1" : "0");
+    }
+
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    public void markAsNotDone() {
+        isDone = false;
+    }
+
+    public String taskInfo() {
+        return "[] [" + getStatusIcon() + "] " + description;
+    }
+
+    public String taskSaveInfo() {
+        return "," + getSavedStatusIcon() + "," + description;
+    }
+
+    public boolean containKeyword(String keyword) {
+        return this.description.contains(keyword);
+    }
```
Perhaps you forgot java docs for these methods?

Donovan9617 reviewed (commented):
Overall, really nice and well organised code. LGTM!

ongwenpin committed:
Add a GUI to Duke

ongwenpin committed:
Merge branch 'branch-Level-10'

LolfoollorS commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,48 @@
+     * @return The message that the deadline has been added and the number of current tasks.
+     */
+    @Override
+    public String execute(TaskList task, Storage storage) {
+        Deadline currDeadline = new Deadline(this.description, this.localDate, this.localTime);
+        task.addTask(currDeadline);
+        return "Got it. I've added this task:\n " + currDeadline.taskInfo()
+                + "\nNow you have"  + task.getTaskSize() + " tasks in the list.";
```
Perhaps you could rename the variable `task` to sound like it is representing a collection of objects.

LolfoollorS commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,42 @@
+     * @return The message that the task has been deleted if successfully deleted.
+     */
+    @Override
+    public String execute(TaskList task, Storage storage) {
+        try {
+            Task taskToDelete = task.getTask(toDelete);
+            task.deleteTask(toDelete);
```
Perhaps you could rename the variable `task` to sound like it is representing a collection of objects.

LolfoollorS commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,42 @@
+     * @return The message that the event has been added and the number of current tasks.
+     */
+    @Override
+    public String execute(TaskList task, Storage storage) {
+        Event currEvent = new Event(this.description, this.duration);
+        task.addTask(currEvent);
```
Perhaps you could rename the variable task to sound like it is representing a collection of objects.

LolfoollorS commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,42 @@
+    @Override
+    public String execute(TaskList task, Storage storage) {
+        ArrayList<Task> matchedTasks = task.findTasks(this.keyword);
+        String listOfMatchedTasks = "Here are the matching tasks in your list\n";
+        for (int i = 0; i < matchedTasks.size(); i++) {
+            listOfMatchedTasks += (i + 1) + ". " + matchedTasks.get(i).taskInfo() + "\n";
+        }
```
Perhaps using a String builder would speed up this process if there are a lot of tasks.

LolfoollorS reviewed (commented):
LGTM. Just some small things to fix. 😉

ongwenpin committed:
Fix code structure and naming convention from PR review

ongwenpin committed:
Add assertion to some commands, parser and storage

ongwenpin committed:
Improve code quality

ongwenpin committed:
Merge pull request #1 from ongwenpin/branch-A-Assertions

Add assertion to some commands, parser and storage

ongwenpin committed:
Merge branch 'master' into branch-A-CodeQuality

ongwenpin committed:
Merge pull request #2 from ongwenpin/branch-A-CodeQuality

Improve code quality

ongwenpin committed:
Set up Continuous Integration

ongwenpin committed:
Add Assertion and improve code quality

Add some assertion to important part of the code where assumption are hold
to any mistake in the code.

Improve code quality by refactoring code such as removing any magic literals,
make happy path more prominent to ensure code readability.

Fix also coding standard based on feedback from PR review.

ongwenpin committed:
Add priority to tasks

ongwenpin committed:
Merge branch 'branch-C-Priority'

ongwenpin committed:
Fix iP code quality feedback and bug from priority

ongwenpin committed:
Improve the GUI

ongwenpin committed:
Give chatbot a unique personality

ongwenpin committed:
Adjust background colour of GUI

ongwenpin committed:
Add a User Guide

ongwenpin committed:
Fix some bug regarding app not exiting

ongwenpin committed:
Update README.md

ongwenpin committed:
Add more automated tests

ongwenpin committed:
Release the product

ongwenpin committed:
Update Ui.png
